### ISSUE TITLE
Android: Vibration still not working as documented

### ISSUE THREAD

**Issue by appfamily** [coronalabs/corona#499]
Originally reported in #483 with sample project , fix committed in #484. However, does not work. Copied comment from #483 :

-------
Does not work fully on my device, using build 3684.

The three "notification" styles work as expected now, each generating a distinct pattern.

System.vibrate() does nothing

The three "impact" types do nothing

Invalid type does nothing (except log)

Android 13, Pixel 4a
-------

**Comment by scottrules44** [comment]
I just tried this on Pixel 6a running Android 13 and I can feel the difference between each impact and `system.vibrate()` works for mine device. I am using 2022.3685

```
local json= require "json"
local widget = require("widget")


local bg = display.newRect( display.contentCenterX, display.contentCenterY, display.actualContentWidth, display.actualContentHeight )
bg:setFillColor( 0,0,1 )


local VibTest1 = widget.newButton({
    label = "Vib Test 1",
    x = display.contentCenterX,
    y = display.contentCenterY-150,
    id = "VibTest1",
    labelColor = { default={ 1, 1, 1 }, over={ 0, 0, 0, 0.5 } },
    onRelease =  function ()
        system.vibrate("impact", "heavy") 
    end
})
local VibTest2 = widget.newButton({
    label = "Vib Test 2",
    x = display.contentCenterX,
    y = display.contentCenterY-50,
    id = "VibTest2",
    labelColor = { default={ 1, 1, 1 }, over={ 0, 0, 0, 0.5 } },
    onRelease =  function ()
        system.vibrate("impact", "medium") 
    end
})
local VibTest3 = widget.newButton({
    label = "Vib Test 3",
    x = display.contentCenterX,
    y = display.contentCenterY+50,
    id = "VibTest3",
    labelColor = { default={ 1, 1, 1 }, over={ 0, 0, 0, 0.5 } },
    onRelease =  function ()
        system.vibrate("impact", "light") 
    end
})
local VibTest4= widget.newButton({
    label = "Vib Test 4",
    x = display.contentCenterX,
    y = display.contentCenterY+150,
    id = "VibTest4",
    labelColor = { default={ 1, 1, 1 }, over={ 0, 0, 0, 0.5 } },
    onRelease =  function ()
        system.vibrate() 
    end
})
```

**Comment by appfamily** [comment]
Thanks. I found the issue. None of the options in your sample code worked on my device. 

However, it worked if I enabled the phone setting "Sounds & Vibration -> Vibration and Haptics -> Touch Feedback".

One can argue either way if Solar2d should or should not respect the "Touch Feedback" system setting. I feel that a user might want vibration in a game, but not on their system in general. Also note that Google Play encourages vibration for UI interaction in games for kids.

In my case I can still use notification vibrations if I want them to always work regardless of phone settings. But they will be heavy and/or feel 'wrong' compared to e.g. "impact light".

My proposed resolution to this is that you expose the ability to create your own waveforms for custom vibrations (VibrationEffect.createWaveform(timings, amplitudes, -1) ?). This would allow developers to create and use any vibration regardless of the phone settings (and instead exposing the setting to enable/disable vibration in-game, just like what you typically do with music or SFX)
